tracer/src/Datadog.Trace/Debugger/SpanCodeOrigin/SpanCodeOriginManager.cs

```diff
@@ -0,0 +1,150 @@
+// <copyright file="SpanCodeOriginManager.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Datadog.Trace.Debugger.Symbols;
+using Datadog.Trace.Logging;
+using Datadog.Trace.VendoredMicrosoftCode.System.Buffers;
+using Datadog.Trace.VendoredMicrosoftCode.System.Collections.Immutable;
+
+namespace Datadog.Trace.Debugger.SpanCodeOrigin
+{
+    internal class SpanCodeOriginManager
+    {
+        private const string CodeOriginTag = "_dd.code_origin";
+
+        private const string FramesPrefix = "frames";
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(SpanCodeOriginManager));
+
+        private static object _globalInstanceLock = new();
+
+        private static bool _globalInstanceInitialized;
+
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring as nullable.
+        private static SpanCodeOriginManager _instance;
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring as nullable.
+
+        private readonly DebuggerSettings _settings = LiveDebugger.Instance?.Settings ?? DebuggerSettings.FromDefaultSource();
+
+        internal static SpanCodeOriginManager Instance =>
+            LazyInitializer.EnsureInitialized(
+                ref _instance,
+                ref _globalInstanceInitialized,
+                ref _globalInstanceLock);
```

Comment:
I can't see a reason to use this instead of something much simpler? 🤔 It solves all the warnings too..

```suggestion
        private readonly DebuggerSettings _settings = LiveDebugger.Instance?.Settings ?? DebuggerSettings.FromDefaultSource();

        internal static SpanCodeOriginManager Instance  { get; } = new();
```

Reply:
Absolutely right. thanks

Reply:
Fix in ccc2d563e115d67c5eb0463aae6030070b923cef